rbac/management/role/definer.py

```diff
@@ -24,13 +24,33 @@
 from django.db import transaction
 from django.utils import timezone
 from management.permission.model import Permission
-from management.role.model import Access, ResourceDefinition, Role
+from management.role.model import Access, ExtRoleRelation, ExtTenant, ResourceDefinition, Role
 
 from api.models import Tenant
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+def _add_ext_relation_if_it_exists(external_relation, role):
+    if not external_relation:
+        return
+    public_tenant = Tenant.objects.get(tenant_name="public")
+    ext_id = external_relation.get("id")
+    ext_tenant_name = external_relation.get("tenant")
+
+    ext_tenant, created = ExtTenant.objects.get_or_create(name=ext_tenant_name, tenant=public_tenant)
```

Comment:
Good question, I was thinking since it is one to one relation, they could just remove that old role and then define a new role. 
If they remove the role, since we have the cascade set for external_relation, the relation would be gone with the role. This would be implemented when we support removing existing system roles.

But that might cause issues if the role is assigned to some group, which will remove the old role but not attach the new one to the group.
I think it would be better to support updating existing role. I will update the code to delete the old relation and create the new relation again.